content/guides/typescript-for-solid.mdx

```diff
@@ -0,0 +1,707 @@
+import { Aside } from "~/components/configurable/Aside";
+
+# TypeScript for Solid
+
+## Overview
+
+[TypeScript](https://www.typescriptlang.org/) is a superset of JavaScript that enhances code reliability and predictability through the introduction of [static types](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html). 
+While JavaScript code can be directly used in TypeScript, the added type annotations in TypeScript provide clearer code structure and documentation, making it more accessible for developers.
+
+By leveraging standard JSX, Solid facilitates seamless TypeScript interpretation. 
+Moreover, with Solid's built-in types for the API heighten accuracy. 
+
+For developers eager to get started, we offer TypeScript templates our [GitHub](https://github.com/solidjs/templates/). 
+
+## Configuring TypeScript
+
+When integrating TypeScript with the Solid JSX compiler, there are crucial settings to make for a seamless interaction:
+
+1. `"jsx": "preserve"` in the `tsconfig.json` retains the original JSX form.
```

Comment:
```suggestion
1. `"jsx": "preserve"` in the `tsconfig.json` retains the original JSX form. Solid's JSX transformation is incompatible with TypeScript's JSX transformation.
```
Not sure if this bit of context is important, if you think it isn't feel free to ignore this suggestion.